File: suco/suco/src/main/java/com/example/suco/dto/sos/tinhieu/UserInfoResponseDTO.java
package com.example.suco.dto.sos.tinhieu;

public class UserInfoResponseDTO {
    private String name;
    private boolean vip;
    private String email;
    private int totalPoints;



    public UserInfoResponseDTO() {}

public UserInfoResponseDTO(String name, String email, boolean vip, int totalPoints) {
    this.name = name;
    this.email = email;
    this.vip = vip;
    this.totalPoints = totalPoints;
}

    public String getName() { return name; }
    public void setName(String name) { this.name = name; }

    public boolean isVip() { return vip; }
    public void setVip(boolean vip) { this.vip = vip; }

    public String getEmail() { return email; }
    public void setEmail(String email) { this.email = email; }

    public int getTotalPoints() { return totalPoints; }
    public void setTotalPoints(int totalPoints) { this.totalPoints = totalPoints; }
    
}

File: CanhBao/app/src/main/java/com/example/canhbao/data/model/sos/tinhieu/UserInfoResponseDTO.kt
package com.example.canhbao.data.model.sos.tinhieu

// 1. DTO thông tin cơ bản
data class UserInfoResponseDTO(
    val name: String?,
    val vip: Boolean?, // Jackson map thuộc tính 'boolean vip' thành key "vip"
    val email: String?
)

// 2. DTO mở rộng (đầy đủ các thuộc tính từ cả lớp cha và lớp con ở Backend)
data class UserMiniDTO(
    val id: String?,
    val name: String?,
    val email: String?,
    val vip: Boolean?,
    val totalPoints: Int?
)